### PR TITLE
fix(Wizard): adds default function to onClose prop

### LINKED ISSUE #2831: PF4: Wizard - missing onClose crashes the wizard
The `onClose` prop is optional but the wizard crashes when it is missing.

We can either:

- keep it optional. In that case multiple null checks have to be added to the wizard.
- change it to required which would be a breaking change

### PR DESCRIPTION
**Pull request by SpyTec**
**What**: This adds a default function to `onClose` so that it no longer crashes the component if left out

**Additional issues**: fixes #2831

cc @suomiy

### CODE REVIEW

**Comment by suomiy** [comment]
+1, but snapshots need to be updated

**Comment by SpyTec** [comment]
Right, thank you! - Snapshots updated

File: packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx [hunk 1]
```diff
@@ -121,7 +121,8 @@ export class Wizard extends React.Component<WizardProps, WizardState> {
     onGoToStep: null as WizardStepFunctionType,
     width: null as string,
     height: null as string,
-    footer: null as React.ReactNode
+    footer: null as React.ReactNode,
+    onClose: () => undefined as () => void
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx]: Nitpick: `() => undefined as any` or `(): any => undefined` to be consistent with the rest of our code.

File: packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx [hunk 2]
```diff
@@ -121,7 +121,8 @@ export class Wizard extends React.Component<WizardProps, WizardState> {
     onGoToStep: null as WizardStepFunctionType,
     width: null as string,
     height: null as string,
-    footer: null as React.ReactNode
+    footer: null as React.ReactNode,
+    onClose: () => undefined as any
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx]: Can we do this instead for consistency?
```suggestion
    onClose: (): any  => undefined 
```

Comment (SpyTec) [reply packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx]: Right now there's 11 results for `): any => undefined` and 22 for `) => undefined as any`. Are we preferring the former?

Comment (dlabaj) [reply packages/patternfly-4/react-core/src/components/Wizard/Wizard.tsx]: We should fix these consistency issues and decide on which format we would like to use.  Could add some tslint rules for things when possible.  I would recommend leaving this for now and fixing some of these consistency issues in a separate issue.

**Pull request merged**